packages/core/src/electron-common/messaging/electron-connection-handler.ts

```diff
@@ -0,0 +1,30 @@
+/********************************************************************************
+ * Copyright (C) 2020 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { ConnectionHandler } from '../../common/messaging/handler';
+
+/**
+ * Name of the channel used with `ipcMain.on/emit`.
+ */
+export const THEIA_ELECTRON_IPC_CHANNEL_NAME = 'theia-electron-ipc';
+
+/**
+ * Electron-IPC-specific connection handler.
+ * Use this if you want to establish communication between the frontend and the electron-main process.
+ */
+export const ElectronConnectionHandler = Symbol('ElectronConnectionHandler');
```

Comment:
I wonder whether we really need it. `ConnectionHandler` belongs to common module meaning that it can be used in any context already. It also does not imply any particular connection.

Is there an issue with using `ConnectionHandler` directly?

Reply:
It was just confused handling a connection handler in the electron main process. I knew these are used in the backend process, and seeing it again in the electron main process somehow gave me a hard time reasoning about what does what. Ultimately you are right, and this is how I used to have this implemented: using the `ConnectionHandler` symbol directly.

Also, I know it is a far stretch, but maybe later we'll have connection handlers for communication between backend and electron main process? In such a case, we cannot use the same `ConnectionHandler` symbol because it would be ambiguous as to which service we want to provide a connection handler to.

TL;DR: I found it confusing, but maybe the current naming isn't much more explanatory. Hopefully the jsdoc might help someone hovering the symbol.

Reply:
ok